djsixpack/djsixpack.py

```diff
@@ -111,13 +111,15 @@ def participate(self, force=None, user_agent=None, ip_address=None, prefetch=Fal
             chosen_alternative = resp['alternative']['name']
         finally:
             if self.local and chosen_alternative:
-                try:
+
+                # Record the bucket in the database if one doesn't already exist.
+                if not SixpackParticipant.objects.filter(unique_attr=self.client_id, experiment_name=experiment_name).exists():
                     SixpackParticipant.objects.get_or_create(unique_attr=self.client_id, experiment_name=experiment_name, bucket=chosen_alternative)
```

Comment:
Anytime you have `get_or_create` that could be accessed through an API (ie. frontend), you will run the risk of a race condition that would result an the `MultipleObjectsReturned` exception. We shouldn't remove the `try`.